nemoguardrails/library/gliner/request.py

```
@@ -74,29 +80,68 @@ async def gliner_request(
     async with aiohttp.ClientSession() as session:
         async with session.post(server_endpoint, json=payload, headers=headers) as resp:
             if resp.status != 200:
                 raise ValueError(f"GLiNER call failed with status code {resp.status}.\nDetails: {await resp.text()}")
 
             try:
-                return await resp.json()
+                raw = await resp.json()
             except aiohttp.ContentTypeError:
                 raise ValueError(
                     f"Failed to parse GLiNER response as JSON. Status: {resp.status}, Content: {await resp.text()}"
                 )
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)